hat/assets/js/apps/Iaso/domains/entities/components/EntitiesQuerybuilder/utils.ts

```diff
@@ -0,0 +1,162 @@
+import { JsonLogicTree } from '@react-awesome-query-builder/mui';
+import { QueryBuilderFields } from 'bluesquare-components';
+import { Form } from 'Iaso/domains/forms/types/forms';
+
+export const borderColor = 'rgba(0, 0, 0, 0.23)';
+export type LogicOperator = 'and' | 'or';
+export type NotState = boolean;
+
+export type ShortForm = {
+    id?: number;
+    form_name?: string;
+    form_id?: string;
+};
+
+export type FormState = {
+    form: ShortForm;
+    logic?: JsonLogicTree;
+    operator?: string;
+    fields?: QueryBuilderFields;
+};
+export const defaultFormState: FormState = {
+    form: {
+        id: undefined,
+        form_name: undefined,
+        form_id: undefined,
+    },
+    logic: undefined,
+    operator: undefined,
+};
+export const operatorButtons = [
+    {
+        key: 'and',
+        label: 'AND',
+        alwaysVisible: false,
+    },
+    {
+        key: 'or',
+        label: 'OR',
+        alwaysVisible: false,
+    },
+];
+export const parseInitialLogic = (logicInput: any, formsList: Form[]) => {
+    let parsedNot = false;
+    let logic = logicInput;
+    if (!logic)
+        return {
+            parsedNot,
+            mainOperator: 'and',
+            parsedFormStates: [],
+        };
+    if (typeof logic === 'string') {
+        try {
+            logic = JSON.parse(logic);
+        } catch {
+            return {
+                parsedNot,
+                mainOperator: 'and',
+                parsedFormStates: [],
+            };
+        }
+    }
+    if (logic['!']) {
+        parsedNot = true;
+        logic = logic['!'];
+    }
+    let mainOperator: LogicOperator = 'and';
+    let formLogics: any[] = [];
+    if (logic.and || logic.or) {
+        mainOperator = logic.and ? 'and' : 'or';
+        formLogics = logic[mainOperator];
+    } else if (logic.some || logic.all || logic.none) {
+        formLogics = [logic];
+    } else {
+        return { parsedNot, mainOperator, parsedFormStates: [] };
+    }
```

Comment:
Goes with the one above. Kinda nitpicking, but I think using `const` i.o reassigning `let` variables is easier to read

```suggestion
if (!logic.and && !logic.or && !logic.some && !logic.all && !logic.none){
    return { parsedNot, mainOperator, parsedFormStates: [] };
}
const formLogics = (logic.and || logic.or) ? logic[mainOperator] : [logic];
```